### PR TITLE
Bootstrap in separate file and output folder create automatically if not exists

### LINKED ISSUE #19: Output folder is not automatically created for the report type `out_coverage`
## Expected Behavior

As a user, I expect the output folder specified in my customized `.coverage.json` (key `output`) to be automatically created for the requested report type `out_coverage`.

## Current Behavior

What really happens is that the output folder is not created for the report types `out_coverage` and the following exception trace is printed to the terminal, warning the user about the impossibility of saving the report results due to inexistent path:

```shell
W20160907-12:49:59.731(2)? (STDERR) /tmp/meteor-test-rundxuvs9/.meteor/local/build/programs/server/packages/lmieulet_meteor-coverage.js:539
W20160907-12:49:59.731(2)? (STDERR)         throw 'failed to write report file: ' + reportPath;                                                            // 17
W20160907-12:49:59.732(2)? (STDERR)         ^
W20160907-12:49:59.732(2)? (STDERR) failed to write report file: /home/user/my-meteor-app/packages/my-package/coverage/report.json
```

This error could be concealed by calling any of the previous two report types before or by manually creating the output folder like in the package `circle.yml` or `.travis.yml` files.

I suppose automatically creating output folder is the expected behavior or not, since it's what happens with all other report types (except `out_text_summary`, which does not work, as I've reported in #20).

## Possible Solution

I think the most efficient way to do this is checking for the existence of the output folder (whether if it's customized or not) once for all reports, since it is usual for them to create an output file (if any of them needs to create additional folders like the HTML reporter, then it should check for those subfolders specifically).

I need someone pointing me in the right direction to make a PR for this. I think `ReportCommon.checkDirectory()` should be invoked (which synchronously creates the folder if cannot stat it) before calling any of the available reporters, but don't know where to best place it.

## Steps to Reproduce (for bugs)

Execute any of the following codes with non-existent output folder:

```shell
spacejam test-packages --driver-package practicalmeteor:mocha --coverage out_coverage packages/my-package
spacejam-mocha --coverage out_coverage packages/my-package
```

## Context

I was trying to change the output folder and try all the implemented/available report types.

## Your Environment

- Versions used:
   - meteor@1.4.1.1
   - lmieulet:meteor-coverage@0.9.4
   - practicalmeteor:mocha@2.4.5_6
   - spacejam: https://github.com/serut/spacejam/tarball/windows-suppport-rc4
- Environment name and version: Node.js v6.3.1
- Operating System and version: Ubuntu Xenial Desktop 16.04

### PR DESCRIPTION
**Pull request by cgalvarez**
closes #19. For details about the enhacement this PR implements, please, read the issue #19.

It should not break previous functionality. Please, anyone interested give it a test :wink:

### CODE REVIEW

File: server/boot.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import path from 'path';
+import mkdirp from 'mkdirp';
+import Conf from './context/conf';
+import Router from './router';
+import Instrumenter from './services/instrumenter';
+
+export default Boot = {
+  startup() {
+    // Create reports output folder if not exists
+    let outputFolder = path.resolve(Conf.COVERAGE_APP_FOLDER, `.${path.sep}${Conf.COVERAGE_EXPORT_FOLDER}`);
```
Comment (serut) [inline server/boot.js]: `path.join(Conf.COVERAGE_APP_FOLDER, Conf.COVERAGE_EXPORT_FOLDER)` can make the job

File: server/boot.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    fs.access(outputFolder, fs.F_OK | fs.R_OK | fs.W_OK, (err) => {
+      err && mkdirp(outputFolder);
```
Comment (serut) [inline server/boot.js]: The folder creation folder can fail, so you need to  
```
try  {
    if (err)
         mkdirp(outputFolder);
} catch {
     console.error ("meteor-coverage failed to create the folder", outputFolder, " while booting :", e)
     Log.error(e.stack); 
}
```

**Comment by serut** [comment]
Nothing more to say, it's nearly perfect ! Good job @cgalvarez !

**Comment by cgalvarez** [comment]
Nice catch with the `mkdirp` exception!

I prefer `path.resolve` because users can use relative paths for `COVERAGE_APP_FOLDER` or the `output` key in the settings, like `../../myCoverageFolder`. `path.join()` simply concatenates the two provided strings, while `path.resolve()` resolves and provides a final absolute path.

If you prefer `path.join()`, I can change that.

**Comment by serut** [comment]
Not really, path.join is not just a concatenation !
```
path.join('/foo', 'bar', 'baz/asdf', 'quux', '..')
// returns '/foo/bar/baz/asdf'
```

**Comment by cgalvarez** [comment]
There you have it, @serut! Updated and rebased. Thanks for teaching me something new about `path.join()` :wink:

**Comment by serut** [comment]
As always, it's perfect ! Thank you @cgalvarez !

**Pull request merged**